### PR TITLE
chore: Add temp fix for prod deployment

### PR DESCRIPTION
**Pull request by tihuan**
### Reviewers
**Functional:** 

**Readability:** 

---

## Changes
- add
- remove
- modify

Add our temp fix for PROD deployment in the file. Also note that STAGING deployment doesn't need the temp fix!

Thank you!

### CODE REVIEW

**Review by metakuni — APPROVED** [review]

File: Dockerfile.processing_image [hunk 1]
```diff
@@ -12,11 +12,12 @@ ENV PIP_NO_CACHE_DIR=1
...
 # The no-op npm touches some files that are the cellxgene install script expects to exist
 RUN git clone https://github.com/chanzuckerberg/single-cell-explorer.git \
- && echo '#!/bin/sh\nmkdir -p build/static && touch build/static/.gitignore build/index.html build/csp-hashes.json' > /usr/bin/npm && chmod 755 /usr/bin/npm \
- && cd single-cell-explorer \
- && make build-for-server-dev \
- && cd .. \
- && rm -rf single-cell-explorer client build/client /usr/bin/npm
+  && echo '#!/bin/sh\nmkdir -p build/static && touch build/static/.gitignore build/index.html build/csp-hashes.json' > /usr/bin/npm && chmod 755 /usr/bin/npm \
+  #  NOTE: Uncomment this line for prod deployment
+  # && cd single-cell-explorer \
```
Comment (tihuan) [inline Dockerfile.processing_image:17]: Actually these two lines lol

File: Dockerfile.processing_image [hunk 2]
```diff
@@ -12,11 +12,12 @@ ENV PIP_NO_CACHE_DIR=1
...
 RUN git clone https://github.com/chanzuckerberg/single-cell-explorer.git \
- && echo '#!/bin/sh\nmkdir -p build/static && touch build/static/.gitignore build/index.html build/csp-hashes.json' > /usr/bin/npm && chmod 755 /usr/bin/npm \
- && cd single-cell-explorer \
- && make build-for-server-dev \
- && cd .. \
- && rm -rf single-cell-explorer client build/client /usr/bin/npm
+  && echo '#!/bin/sh\nmkdir -p build/static && touch build/static/.gitignore build/index.html build/csp-hashes.json' > /usr/bin/npm && chmod 755 /usr/bin/npm \
+  && cd single-cell-explorer \
+  #  NOTE: Uncomment this line for prod deployment
+  #  && make build-for-server-dev \
```
[the remark this answers is not in the document]
Comment (atolopko-czi) [reply Dockerfile.processing_image]: QFMU: What happens if this were to run during a prod deploy? Why does it need to be commented out?

File: Dockerfile.processing_image (hunk 1 above)
Comment (atolopko-czi) [reply Dockerfile.processing_image:17]: also the `cd ..` line below?

Comment (atolopko-czi) [reply Dockerfile.processing_image:17]: actually I think the next line will be interpreted as a new command and the `&&` will cause it to fail, being an invalid start of a command.  The trailing `\` need to be continuous on every line that is connected; the comments break that.

Comment (tihuan) [reply Dockerfile.processing_image:17]: ah interesting, the `cd ..` was already there, so maybe I need to add another `cd ..` and comment it out 🤔 ?

Comment (tihuan) [reply Dockerfile.processing_image:17]: oh wait, I confused myself 😭 The commented code should be checking out `staging` ahaha

File: Dockerfile.processing_image [hunk 3]
```diff
@@ -12,11 +12,13 @@ ENV PIP_NO_CACHE_DIR=1
...
 RUN git clone https://github.com/chanzuckerberg/single-cell-explorer.git \
- && echo '#!/bin/sh\nmkdir -p build/static && touch build/static/.gitignore build/index.html build/csp-hashes.json' > /usr/bin/npm && chmod 755 /usr/bin/npm \
- && cd single-cell-explorer \
- && make build-for-server-dev \
- && cd .. \
- && rm -rf single-cell-explorer client build/client /usr/bin/npm
+  && echo '#!/bin/sh\nmkdir -p build/static && touch build/static/.gitignore build/index.html build/csp-hashes.json' > /usr/bin/npm && chmod 755 /usr/bin/npm \
+  && cd single-cell-explorer \
+  #  NOTE: Uncomment this line for prod deployment
+  # && git checkout staging \
```
Comment (tihuan) [inline Dockerfile.processing_image:18]: Sorry this should be the right code this time 🤦 

Screen Shot 2021-11-03 at 12 43 19 PM

**Comment by metakuni** [comment]
I think we no longer need this workaround. See https://czi-sci.slack.com/archives/C0244PQK934/p1635968808084300?thread_ts=1635960252.071500&cid=C0244PQK934

**Comment by tihuan** [comment]
Awesome! Thanks Kuni! Closing the PR now it's irrelevant 🎉

**Pull request closed without merging**